src/ray/object_manager/object_manager.cc

```diff
@@ -478,28 +527,24 @@ void ObjectManager::Push(const ObjectID &object_id, const NodeID &client_id) {
                    << ", total data size: " << data_size;
 
     UniqueID push_id = UniqueID::FromRandom();
-    for (uint64_t chunk_index = 0; chunk_index < num_chunks; ++chunk_index) {
-      rpc_service_.post([this, push_id, object_id, owner_address, client_id, data_size,
-                         metadata_size, chunk_index, rpc_client]() {
-        auto st = SendObjectChunk(push_id, object_id, owner_address, client_id, data_size,
-                                  metadata_size, chunk_index, rpc_client);
-        if (!st.ok()) {
-          RAY_LOG(WARNING) << "Send object " << object_id << " chunk failed due to "
-                           << st.message() << ", chunk index " << chunk_index;
-        }
-      });
-    }
+    push_manager_->StartPush(push_id, num_chunks, [=](int64_t chunk_id) {
+      SendObjectChunk(push_id, object_id, owner_address, client_id, data_size,
```

Comment:
I wanted to isolate the dependency here so PushManager doesn't depend on object manager internals. This separation makes it easier to unit test.

Reply:
The follow-up PR is changing PushManager to be handling the callbacks for individual chunks. How about we wait on that one for cleaning this up?

For now, I think this design gives the best isolation between the components making it much easier to test PushManager.

Reply:
Sounds good. I guess we will need to mock out the RPC client in the next PR, but I think it's worthwhile to do so that the PushManager is self-contained.